### ISSUE TITLE
fasterq-dump can delete everything in the temp directory through symbolic link

### ISSUE THREAD

**Issue by mengzhou** [ncbi/sra-tools#163]
I am using `fasterq-dump` v2.9.2 (pre-built binary for ubuntu 64-bit), and it gave me some error. I noticed that the temp directory `fast.tmp` was not created during this run, so I wanted to specifically set this directory to see what's going on during this run. Then I used `-t $PWD/` to make sure it is writing temporary files to where I know. The consequence is that indeed `fasterq-dump` writes to this current directory, but before writing any temp files it erased *everything* in this directory.

This problem occurs when `-t` points to a link, e.g. if you have `/home/user/data -> /home/user/true_location/`, and set `-t ~/data/`, then everything in this directory will be deleted before any temp file is generated.

This is scary. I just lost 4TB of data. Fortunately those were not critical data and I can re-generate those. But please fix this problem, and I hope nobody else has to experience this error.

**Comment by kwrodarmer** [comment]
Thank you for the report. We will address it as quickly as possible.

**Comment by altairwei** [comment]
I have noticed that `fasterq-dump` will delete all files in output directory before dumping.

Here is the command that I acturelly excuted:

```shell
fasterq-dump -3 -p -e 8 -o data/reads/SRX3200190 data/sra/SRX3200190.sra
```

Any files in `data/reads/` will be deleted. Is it a feature or bug ? How can I disable this behaviour ?

@kwrodarmer

**Comment by sbrb** [comment]
> I have noticed that `fasterq-dump` will delete all files in output directory before dumping.
> 
> Here is the command that I acturelly excuted:
> 
> ```shell
> fasterq-dump -3 -p -e 8 -o data/reads/SRX3200190 data/sra/SRX3200190.sra
> ```
> 
> Any files in `data/reads/` will be deleted. Is it a feature or bug ? How can I disable this behaviour ?
> 
> @kwrodarmer

This is obviously a bug, and actually a critical one. The original question asks about 'temp' directory, but it's the same behavior with 'output'.

**Comment by klymenko** [comment]
@mengzhou , was your issue resolved?

**Comment by mengzhou** [comment]
@klymenko I have been avoiding the use of this `-t` option because of potential data loss. Not sure if it has been fixed in the latest version of sratools. I wonder if anyone has tested it recently.

**Comment by klymenko** [comment]
AFAIK it was fixed.
@wraetz , please confirm.

**Comment by EmilyB17** [comment]
I just had this issue. Fastq-dump was giving an error attributed to GNU parallel so I was testing faster-q dump. I deleted everything in the output directory. Over 20 TB of data gone in a few seconds. Hopefully this is something that I did wrong and not a bug that others are experiencing as well. @klymenko 

`fasterq-dump ERR2749132 --split-files --outfile ~/scratch/test`

**Comment by wraetz** [comment]
Which version of fasterq-dump are you running? Please check by running 'fasterq-dump --version'

**Comment by EmilyB17** [comment]
"fasterq-dump" version 2.10.9



Emily Bean Van Syoc, M.S.

Integrative & Biomedical Physiology,
Clinical & Translational Sciences Dual-Title PhD Program
Huck Institute of Life Sciences
***@***.***
Pronouns: she/her/hers

________________________________

**Comment by wraetz** [comment]
The problem is fixed in the current version 2.11.0. Please upgrade your installation. (https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.11.0/)
I conducted an experiment reproducing your commandline on my machine with the current version of the tool. It did not empty the ~/scratch directory.

**Comment by EmilyB17** [comment]
Great, glad to hear it is fixed. Maybe you can deprecate the older version or add a very clear warning to the man page to help others not lose 20 TB of their data 🙂

Emily Bean Van Syoc, M.S.

Integrative & Biomedical Physiology,
Clinical & Translational Sciences Dual-Title PhD Program
Huck Institute of Life Sciences
***@***.***
Pronouns: she/her/hers

________________________________
